test_typing_inspect.py

```diff
@@ -165,39 +237,71 @@ def test_last_args(self):
+        if WITH_CLASSVAR:
+            self.assertEqual(get_last_args(ClassVar[int]), (int,))
         self.assertEqual(get_last_args(Union[T, int]), (T, int))
-        self.assertEqual(get_last_args(Iterable[Tuple[T, S]][int, T]), (int, T))
-        self.assertEqual(get_last_args(Callable[[T, S], int]), (T, S, int))
-        self.assertEqual(get_last_args(Callable[[], int]), (int,))
+        base_tp = Iterable[Tuple[T, S]]
+        try:
+            tp = base_tp[int, T]
+        except TypeError:
+            # python 3.5.3- - TypeError: Cannot change parameter count from 1 to 2
+            pass
+        else:
+            self.assertEqual(get_last_args(tp), (int, T))
+        if LEGACY_TYPING:
+            self.assertEqual(get_last_args(Callable[[T, S], int]), (T, S))
+            self.assertEqual(get_last_args(Callable[[], int]), ())
+        else:
+            self.assertEqual(get_last_args(Callable[[T, S], int]), (T, S, int))
+            self.assertEqual(get_last_args(Callable[[], int]), (int,))
 
     @skipIf(NEW_TYPING, "Not supported in Python 3.7")
     def test_args(self):
-        T = TypeVar('T')
-        self.assertEqual(get_args(Union[int, Tuple[T, int]][str]),
-                         (int, (Tuple, str, int)))
-        self.assertEqual(get_args(Union[int, Union[T, int], str][int]),
-                         (int, str))
+        if EXISTING_UNIONS_SUBSCRIPTABLE:
+            T = TypeVar('T')
+            self.assertEqual(get_args(Union[int, Tuple[T, int]][str]),
+                             (int, (Tuple, str, int)))
+            self.assertEqual(get_args(Union[int, Union[T, int], str][int]),
+                             (int, str))
         self.assertEqual(get_args(int), ())
 
     def test_args_evaluated(self):
         T = TypeVar('T')
-        self.assertEqual(get_args(Union[int, Tuple[T, int]][str], evaluate=True),
-                         (int, Tuple[str, int]))
-        self.assertEqual(get_args(Dict[int, Tuple[T, T]][Optional[int]], evaluate=True),
-                         (int, Tuple[Optional[int], Optional[int]]))
-        self.assertEqual(get_args(Callable[[], T][int], evaluate=True), ([], int,))
+        if EXISTING_UNIONS_SUBSCRIPTABLE:
+            self.assertEqual(get_args(Union[int, Tuple[T, int]][str], evaluate=True),
+                             (int, Tuple[str, int]))
+        base_tp = Dict[int, Tuple[T, T]]
+        try:
+            tp = base_tp[Optional[int]]
+        except TypeError:
+            # python 3.5.3 - TypeError: Cannot change parameter count from 2 to 1
+            pass
+        else:
+            self.assertEqual(get_args(tp, evaluate=True),
+                             (int, Tuple[Optional[int], Optional[int]]))
+
+        base_tp = Callable[[], T]
+        try:
+            tp = base_tp[int]
+        except TypeError:
+            # python 3.5.3- - TypeError: This Callable type is already parameterized.
+            pass
```

Comment:
These two can again use the same constant set above.

Reply:
Not the same constant for readability, but yes